_pytest/unittest.py

```diff
@@ -134,6 +134,7 @@ def stopTest(self, testcase):
         pass
 
     def runtest(self):
+        setattr(self._testcase, "__name__", self.name)
```

Comment:
I think this would be better placed in the `setup` method instead. Also, please include a comment explaining why this is being added and mentioning #1169 for future reference.